trie/trienode/node.go

```diff
@@ -153,6 +153,15 @@ func (set *NodeSet) Size() (int, int) {
 	return set.updates, set.deletes
 }
 
+// HashSet returns a set of trie nodes keyed by node hash.
+func (set *NodeSet) HashSet() map[common.Hash][]byte {
+	ret := make(map[common.Hash][]byte)
```

Comment:
We could preallocate here 
```suggestion
	ret := make(map[common.Hash][]byte, len(set.Nodes)
```